app/form_models/publishers/job_listing/applying_for_the_job_form.rb

```diff
@@ -24,4 +29,12 @@ def override_enable_job_applications_for_local_authority!
 
     self.enable_job_applications = false
   end
+
+  # If a Publisher publishes a vacancy for a job role that does not allow enabling job applications
+  # but then changes the job role to one that does, we don't want them to be able to allow applications
+  # but we do want the validations for the enable_job_applications field to pass to prevent an error
+  # from being displayed on the review page.
```

Comment:
Is it specifically about job roles? I thought it was just that
1) we don't want `enable_job_applications` to be able to change after a job is published
but
2) we do want the step to be valid

Reply:
The step only becomes invalid when the job role is changed to a job role that allows enabling job applications as the validation is only run if `vacancy.allow_enabling_job_applications?`

Reply:
Isn't validation run whenever a hiring staff tries to edit a listing?

Reply:
Yeah it is. But it's only invalid when they change job roles from one which doesn't accept applications to one that does.

If they select "Teaching assistant" for example, the field for enable_job_applications is not rendered because this returns false:
```
def allow_enabling_job_applications?
  %w[teacher leadership sendco].include?(main_job_role)
end
```
This then means that `enable_job_applications` is nil. The validation is not run because the condition (which uses the above method - see app/models/form_models/publishers/job_listing/applying_for_the_job.html.slim) evaluates to false.

If they then edit the vacancy and change the job role, when all of the steps are validated, this validation is now run. This is because `allow_enabling_job_applications?` now evaluates to true. The validation then fails as nil is not in `[true, false, "true", "false"]`

Reply:
Sorry, which validation is not run?

Reply:
Oh, I found it:

```
   validates :enable_job_applications, inclusion: { in: [true, false, "true", "false"] }, if: proc { vacancy.allow_enabling_job_applications? }
```

Reply:
That's the one!